sqllogictest/src/runner.rs

```diff
@@ -1090,6 +1091,9 @@ impl<D: AsyncDB, M: MakeConnection<Conn = D>> Runner<D, M> {
 
     /// accept the tasks, spawn jobs task to run slt test. the tasks are (AsyncDB, slt filename)
     /// pairs.
+    // TODO: This is not a good interface, as the `make_conn` passed to `new` is unused but we
+    // accept a new `conn_builder` here. May change `MakeConnection` to support specifying the
+    // database name in the future.
```

Comment:
I don't understand the comment here. Where's the `make_conn` passed to `new`?

Reply:
Oh, do you mean `self.conn`?

Reply:
Ok, so it seems `self.conn` is used to `CREATE DATABASE`, but we need new conns for individual databases.

Reply:
The idea of `conn_builder` here has something common with `make_conn` passed to `new`, but they're not compatible and seem to be kind of redundant to me. For example,

- The first arg is a `host` randomly picked from the `hosts`, which is strange. If the callers want load balancing between hosts, they are allowed to capture a slice of `hosts` and do random picking in the body of `make_conn` or `conn_builder`.

- The procedures to establish a connection to the database in `make_conn` and `conn_builder` are likely to the same, except for targeting different databases. We should unify them by changing the definition of `MakeConnection`. Only using the one passed in `new` to create databases in parallel mode does not seem to be a good design.